### ISSUE TITLE
cannot send post method in "request" function from Axios instance

### ISSUE THREAD

**Issue by alpancs** [axios/axios#723]
When I try to send POST method using this code:
```
const axios = require('axios')
let config = {url: 'http://{{hostname}}', method: 'post'}
let instance = axios.create(config)
instance.request()
```
It send GET method instead of POST.

I think this line https://github.com/mzabriskie/axios/blob/master/lib/core/Axios.js#L37 should be: `config = utils.merge(defaults, this.defaults, config);`

**Comment by chengluyu** [comment]
I've got the same problem. Here is the code.

![2017-02-26 6 39 10](https://cloud.githubusercontent.com/assets/2239547/23338983/3e935172-fc53-11e6-8c04-9a26015cabbc.jpg)

The `login` function uses [request](https://github.com/request/request) module, which worked well. However, the `axiosLogin` function, which uses axios, always failed. Both of them were applied with same arguments.

I'm not sure whether that line matters.

**Comment by alpancs** [comment]
I have tested by modifying this line https://github.com/mzabriskie/axios/blob/master/lib/core/Axios.js#L37 into `config = utils.merge(defaults, this.defaults, config);`. It works, but no permission to push my branch.

**Comment by CameronK** [comment]
I also have this issue. It seems to me that `instance.request()` _should_ use the method defined in *options*, falling back to *GET* if necessary.

**Comment by mightym** [comment]
I even have this with a normal POST request:
```
    axios.defaults.headers.common['X-CSRFToken'] = getCookie('csrftoken');
    return axios({
      method: 'post',
      url: url,
      data: {
        link: link
      }
    })
    .then(function (response) {
      console.log(response);
    })
    .catch(function (error) {
      console.log(error);
    });
```

Its really strange. Chrome/Safari tell me that it is a XHR GET request.

**Comment by rubennorte** [comment]
Right, it seems we should move the default method to defaults:

```javascript
config = utils.merge(defaults, this.defaults, { method: 'get' }, config);
```

Have you considered creating a PR?

Thanks for reporting.

**Comment by humiston** [comment]
Is there currently any workaround for this? I'm getting the same thing regardless of what I do ( `axios({metodh: 'POST', ...})` and `axios.post(...)`).

**Comment by humiston** [comment]
I am experiencing this as well. When attempting to use both `axios({method: 'post', ...)` and `axios.post(...)`, I see in the response that a GET request is being fired. This pretty much makes axios worthless to me.

**Comment by emilyemorehouse** [comment]
Fixed in #1342
